ros2controlcli/doc/userdoc.rst

```diff
@@ -74,6 +75,45 @@ Example output:
     joint_trajectory_controller/JointTrajectoryController                  controller_interface::ControllerInterface
 
 
+list_hardware_components
+------------------------
+
+.. code-block:: console
+
+    $ ros2 control list_hardware_components -h
+    usage: ros2 control list_hardware_components [-h] [--spin-time SPIN_TIME] [-s] [-c CONTROLLER_MANAGER] [--include-hidden-nodes]
+
+    Output the list of available hardware components
+
+    options:
+    -h, --help            show this help message and exit
+    --spin-time SPIN_TIME
+                            Spin time in seconds to wait for discovery (only applies when not using an already running daemon)
+    -s, --use-sim-time    Enable ROS simulation time
+    -c CONTROLLER_MANAGER, --controller-manager CONTROLLER_MANAGER
+                            Name of the controller manager ROS node
+    --include-hidden-nodes
```

Comment:
Can you add `-v` flag as we have in the `list_controllers` verb for this verbose output and per default we just list components and show their status (check—how is this done for controllers).

Does this make sense?

Reply:
It would also be cool that the name is colored depending on their status

Reply:
Just to clarify, the command and state interfaces would only be printed with the `-v` option? It looks like `-v` is by default `true` in `list_controller.py`

And for color based on status, do you mean to color based on state of the hardware component? Based on possible states [here](https://github.com/ros2/rcl_interfaces/blob/humble/lifecycle_msgs/msg/State.msg), what should the color scheme be for the different states?

Reply:
It should not be true per default. Per default, you get only controller's names and their states. 

> Just to clarify, the command and state interfaces would only be printed with the -v option?

Yes, this is correct.

> And for color based on status, do you mean to color based on state of the hardware component? Based on possible states [here](https://github.com/ros2/rcl_interfaces/blob/humble/lifecycle_msgs/msg/State.msg), what should the color scheme be for the different states?

You can copy color scheme from the `list_controllers`. I don't know it by hearth, but it should be clear from the implementation.

Reply:
Added the verbose option, but did not update color scheme... still confused about color scheme to apply. `list_controllers` doesn't have any color scheme that I can follow and this open PR says `list_controllers` should have colorized output.

The colors in `list_hardware_interfaces` are blue for `available claimed`, cyan for `available unclaimed` and yellow/warning for `unavailable unclaimed`. Should I apply this scheme?

Reply:
Guys, can we please defer the decision on exact colours to a follow-up PR?
The functionality is already great and I'd rather have it early and with the wrong colours.

Motivation in figure below:
![2uew66](https://user-images.githubusercontent.com/3524577/215818713-88a90796-d8f5-4d21-ab82-68776f01b243.jpg)

Reply:
A new PR sounds fine to me. There was another comment that the defined colors should be broken out of their current location so could combine with that.

Reply:
Let's keep going then!